static/src/javascripts/projects/common/modules/commercial/commercial-features.ts

```diff
@@ -1,9 +1,35 @@
+import { log } from '@guardian/libs';
 import defaultConfig from '../../../../lib/config';
 import { getBreakpoint } from '../../../../lib/detect';
 import { isUserLoggedIn } from '../identity/api';
 import userPrefs from '../user-prefs';
 import { isAdFreeUser } from './user-features';
 
+/**
+ * Log the reason why adverts are disabled
+ *
+ * @param trueConditions - normally true conditions, log if false
+ * @param falseConditions - normally false conditions, log if true
+ */
+function adsDisabledLogger(
+	trueConditions: Record<string, boolean>,
+	falseConditions: Record<string, boolean>,
+): void {
+	const noAdsLog = (condition: string, value: boolean): void =>
+		log(
+			'commercial',
+			`Adverts are not shown because ${condition} = ${value.toString()}`,
+		);
+
+	Object.entries(trueConditions).forEach(
+		([condition, value]) => !value && noAdsLog(condition, value),
+	);
+
+	Object.entries(falseConditions).forEach(
+		([condition, value]) => value && noAdsLog(condition, value),
+	);
```

Comment:
If you have side-effects, I think a `for … of` loop is more explicit.

```ts
for (const [condition, value] of Object.entries(falseConditions)) {
  if(value) noAdsLog(condition, value);
}

```

Reply:
cool, okay

Reply:
Haha just saw that it was what you did first! While I find this approach more explicit, you are free to choose `forEach` instead if it does the same.

Reply:
Hehe me and Zeke were trying to decide which way to use, neither of us were that picky!